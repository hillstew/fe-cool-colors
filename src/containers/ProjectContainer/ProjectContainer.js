import React, { Component } from 'react';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';
import ProjectCard from '../ProjectCard/ProjectCard';

export class ProjectContainer extends Component {
  render() {
    const { projects, palettes } = this.props;
    const projectCards = projects.map(project => {
      const matchingPalettes = palettes.filter(palette => {
        return palette.project_id === project.id;
      });

      return (
        <ProjectCard matchingPalettes={matchingPalettes} project={project} key={project.id} />
      );
    });

    return <div className="ProjectContainer">{projectCards}</div>;
  }
}

export const mapStateToProps = state => ({
  projects: state.projects,
  palettes: state.palettes
});

export const mapDispatchToProps = dispatch => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProjectContainer);
