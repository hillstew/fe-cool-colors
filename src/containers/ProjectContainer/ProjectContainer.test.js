import React from 'react';
import { ProjectContainer, mapStateToProps } from './ProjectContainer';
import { shallow } from 'enzyme';

describe('ProjectContainer', () => {
  let wrapper;
  let mockProjects = [{ name: 'Movie-Tracker', id: 12 }];
  let mockPalettes = [{ name: 'blue-palette', project_id: 12 }];

  beforeEach(() => {
    wrapper = shallow(
      <ProjectContainer projects={mockProjects} palettes={mockPalettes} />
    );
  });

  it('should match the snaphot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with projects palettes', () => {
      const mockState = {
        projects: mockProjects,
        palettes: mockPalettes,
        extraKey: true
      };
      const expected = {
        projects: mockProjects,
        palettes: mockPalettes
      };

      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });
});
