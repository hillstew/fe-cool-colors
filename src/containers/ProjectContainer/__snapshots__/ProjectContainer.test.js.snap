// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer should match the snaphot 1`] = `
<div
  className="ProjectContainer"
>
  <Connect(ProjectCard)
    key="12"
    matchingPalettes={
      Array [
        Object {
          "name": "blue-palette",
          "project_id": 12,
        },
      ]
    }
    project={
      Object {
        "id": 12,
        "name": "Movie-Tracker",
      }
    }
  />
</div>
`;
